janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -705,11 +730,19 @@ public void mutate(Map<String, Map<String, IndexMutation>> mutations, KeyInforma
                             requestByStore.add(ElasticSearchMutation.createDeleteRequest(indexStoreName, storeName,
                                     documentId));
                         } else {
-                            final String script = getDeletionScript(information, storeName, mutation);
-                            final Map<String,Object> doc = compat.prepareScript(script).build();
+                            Map doc;
+                            if (compat.supportsParams()) {
+                                List<Map<String, Object>> params = getParameters(information.get(storeName),
+                                        mutation.getDeletions(), true);
+                                doc = compat.prepareScript(PARAMETERIZED_DELETION_SCRIPT, params).build();
+                                log.trace("Deletion script {} with params {}", PARAMETERIZED_DELETION_SCRIPT, params);
+                            } else {
+                                final String script = getDeletionScript(information, storeName, mutation);
```

Comment:
If you move the logic from `ES6Compat` into `AbstractESCompat` than `supportsParams` parameter isn't needed (because it will be always `true`). That is why, the logic in `else` statement won't be needed also.

Reply:
done